zipline/data/minute_bars.py

```diff
@@ -706,3 +698,16 @@ def unadjusted_window(self, fields, start_dt, end_dt, sids):
                 out *= self._ohlc_inverse
             results.append(out)
         return results
+
+    def unadjusted_window_for_minutes(self, fields, minutes, sids):
+        start_dt = minutes[0]
+        end_dt = minutes[-1]
+        results = self.unadjusted_window(fields, start_dt, end_dt, sids)
+        if len(results[0][0]) != len(minutes):
+            start_idx = self._find_position_of_minute(start_dt.value)
+            end_idx = self._find_position_of_minute(end_dt.value)
+            data_minutes = self._minute_value_index[start_idx:end_idx + 1]
+            indices = np.in1d(data_minutes, minutes.asi8)
```

Comment:
actually, there's an `assume_unique` flag that I think we can set to True, which would make this faster: https://github.com/numpy/numpy/blob/v1.10.1/numpy/lib/arraysetops.py#L318